### PR TITLE
Feature/psr4

### PR DESCRIPTION
**Pull request by chilimatic**
this should be psr4 compatible now ..... pls review and test it.

### CODE REVIEW

**Comment by andreas83** [comment]
Wohoo looks really good.
I will test and merge it as soon as possible.

**Comment by andreas83** [comment]
i get the following error:
Fatal error: Class '\SocialNetwork\app\controller\DataController' not found in SocialNetwork/routing.php on line 105

**Comment by chilimatic** [comment]
you triggered a composer update ? so the autoloader is registered ?

**Comment by andreas83** [comment]
sorry my fault !

**Comment by andreas83** [comment]
okay looks like most parts works again, except:
 Call to undefined function SocialNetwork\app\lib\transformer\string\html\YoutubeTransformer() in <b>/SocialNetwork/app/lib/transformer/TransformerFactory.php</b> on line <b>63</b><br />

**Comment by andreas83** [comment]
Found another one:
Fatal error: Call to a member function fetchALL() on boolean in /home/lissi/dev/SocialNetwork/app/lib/BaseModel.php on line 5

Reason : namespace is now included in $this->table 

SELECT \* FROM SocialNetwork\app\model\Content WHERE id = 356

**Comment by chilimatic** [comment]
seperated the object name from the table name should work now

**Comment by andreas83** [comment]
Yepp the getSource method works!
Any idea whats wrong with the TransformerFactory ?

**Comment by andreas83** [comment]
mhm another question, is the app/lib/AutoLoader.php missing ?

**Comment by chilimatic** [comment]
yes since it's psr4 the composer autoloader can load the classes. But you can take the file from 

https://github.com/andreas83/SocialNetwork/pull/60/commits/b3a4c33ace2deeb4ac837521f4f6eb1770662147

and build your own psr4 autoloader out of it  [it just misses the prefix and the basedir map which can be easily added]. 

But since this project is heavily dependent of composer I don't really see the benefit.

**Comment by andreas83** [comment]
The reason why i asked was, because i thought its required since you added the 
require_once 'app/lib/AutoLoader.php'; to some files....

I also see no reason for a separate auto loader since it can be done via autoloader from composer

Found some more issues
- [x] Fatal error: Class 'SocialNetwork\app\lib\User' not found in /home/lissi/dev/SocialNetwork/app/lib/Helper.php on line 82
- [x] Call to undefined function SocialNetwork\app\lib\transformer\string\html\YoutubeTransformer() in /SocialNetwork/app/lib/transformer/TransformerFactory.php on line 63

**Comment by chilimatic** [comment]
lol :D andi !!! <3 I forgot the "new" keyword :D sorry ....
Call to undefined function SocialNetwork\app\lib\transformer\string\html\YoutubeTransformer()

**Comment by andreas83** [comment]
you are my hero ! :kiss: 

shame on me, i should be able to fix such bugs in minutes

**Comment by andreas83** [comment]
makeStatic and make looks 99% identical (possible code smell ?)

**Comment by chilimatic** [comment]
ah yes thx.

File: app/lib/transformer/TransformerFactory.php [hunk 1]
```diff
@@ -50,19 +50,7 @@ public static function makeStatic($name)
...
-        if (isset(self::$runtimeCache[$name])) {
-            return self::$runtimeCache[$name];
-        }
-
-        if (!$this->classExists(__NAMESPACE__, $name)) {
-            return null;
-        }
-        $className = __NAMESPACE__ . '\\' . $name;
-
-        self::$runtimeCache[$name] = new $className();
```
Comment (andreas83) [inline app/lib/transformer/TransformerFactory.php]: Now makeStatic does not create a _new_ instance which was the 1 percent difference

**Comment by andreas83** [comment]
http://my url /api/metadata/?url=https://www.youtube.com/watch?v=PUiaKjVpP8M

For some very unknown reasons VideoTransformerTrait::transform never gets called

**Comment by chilimatic** [comment]
sry, fixed :)

**Comment by andreas83** [comment]
wohooo  looks good, i will merge everything to upstream

**Pull request merged**